### DISCUSSION TITLE
Sending ResetPasswordMail across different panels

### DISCUSSION THREAD

**Discussion by hristijans** [filamentphp/filament#14893] [Help]
I have two panels: admin (the default one) and user.

The idea is from the Admin panel I want to create new user and send reset password link after user is created. The issue I am having is when the email is sent,  reset password link contains admin (default panel link) in the url. If I try to remove the admin part from the url I am getting an error Invalid Signature.

Is this even possible, if so how to send reset password mail/ notification? 

`
protected function sendEmailVerificationNotification(Model $user): void
 {
        if (! $user instanceof MustVerifyEmail) {
            return;
        }

        if ($user->hasVerifiedEmail()) {
            return;
        }

        if (! method_exists($user, 'notify')) {
            $userClass = $user::class;

            throw new Exception("Model [{$userClass}] does not have a [notify()] method.");
        }
        
        $token = BrokerPassword::broker()->createToken($user);

        $notification = app()->make(ResetPassword::class, ['token' => $token]);

        $notificationUrl = Filament::getResetPasswordUrl($token, $user);

        $notificationUrl = Str::replace('admin/', '', $notificationUrl);
        
        $notification->url = $notificationUrl;

        $user->notify($notification);
    }
`

**Comment by leandrocfe** [comment]
Could you share the code you are using?

**Reply by hristijans** [reply]
I updated the question with the snippet where I try to send reset password mail.

**Comment by hristijans** [comment]
I solved it by setting the panel I need 

```
    Filament::setCurrentPanel(Filament::getPanel('mycustompanelid'));
    $notificationUrl = Filament::getResetPasswordUrl($token, $user);
```

**Comment by hristijans** [comment]
Before generating the url you can set the current panel

`Filament::setCurrentPanel(Filament::getPanel('mycustompanelid'));`
